lib/collection/variable-scope.js

```diff
@@ -126,13 +134,26 @@ _.assign(VariableScope.prototype, /** @lends VariableScope.prototype */ {
     },
 
     /**
-     * Fetches a variable from the current scope.
+     * Fetches a variable from the current scope or from parent scopes if present.
      *
-     * @param {String} key - The name of the variable to set.
-     * @returns {*} The value of the specified variable in the current context.
+     * @param {String} key - The name of the variable to get.
+     * @returns {*} The value of the specified variable across scopes.
      */
     get: function (key) {
         var variable = this.values.one(key);
+
+        // if a variable does not exist in local scope, we search all the layers and find the first occurence.
+        if (!variable) {
+            _.forEach(this.layers, function (layer) {
+                variable = layer.one(key);
+
+                // stop at the first occurence
+                if (variable && variable.key === key) {
```

Comment:
why `variable.key === key`?

1. `layer.one(key)` already returns based on key
2. if underlying list as case-insensitivity configured, this might break.